kubectl-plugin/pkg/cmd/create/create_cluster.go

```diff
@@ -0,0 +1,160 @@
+package create
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ray-project/kuberay/kubectl-plugin/pkg/util/generation"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	"github.com/ray-project/kuberay/kubectl-plugin/pkg/util/client"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+type CreateClusterOptions struct {
+	configFlags    *genericclioptions.ConfigFlags
+	ioStreams      *genericclioptions.IOStreams
+	clusterName    string
+	rayVersion     string
+	image          string
+	headCPU        string
+	headMemory     string
+	workerGrpName  string
+	workerCPU      string
+	workerMemory   string
+	workerReplicas int32
+	dryRun         bool
+}
+
+var (
+	createClusterLong = templates.LongDesc(`
+		Creates Ray Cluster from inputed file or generate one for user.
+	`)
+
+	createClusterExample = templates.Examples(`
+		# Creates Ray Cluster from flags input
+		kubectl ray create cluster sample-cluster --ray-version 2.39.0 --image rayproject/ray:2.39.0 --head-cpu 1 --head-memory 5Gi --worker-grp-name worker-group1 --worker-replicas 3 --worker-cpu 1 --worker-memory 5Gi
+	`)
+)
+
+func NewCreateClusterOptions(streams genericclioptions.IOStreams) *CreateClusterOptions {
+	return &CreateClusterOptions{
+		configFlags: genericclioptions.NewConfigFlags(true),
+		ioStreams:   &streams,
+	}
+}
+
+func NewCreateClusterCommand(streams genericclioptions.IOStreams) *cobra.Command {
+	options := NewCreateClusterOptions(streams)
+	cmdFactory := cmdutil.NewFactory(options.configFlags)
+
+	cmd := &cobra.Command{
+		Use:          "cluster [CLUSTERNAME]",
+		Short:        "Create Ray Cluster resource",
+		Long:         createClusterLong,
+		Example:      createClusterExample,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := options.Complete(cmd, args); err != nil {
+				return err
+			}
+			if err := options.Validate(); err != nil {
+				return err
+			}
+			return options.Run(cmd.Context(), cmdFactory)
+		},
+	}
+
+	cmd.Flags().StringVar(&options.rayVersion, "ray-version", "2.39.0", "Ray Version to use in the Ray Cluster yaml. Default to 2.39.0")
+	cmd.Flags().StringVar(&options.image, "image", options.image, "Ray image to use in the Ray Cluster yaml")
+	cmd.Flags().StringVar(&options.headCPU, "head-cpu", "500m", "Number of CPU for the ray head. Default to 500m")
+	cmd.Flags().StringVar(&options.headMemory, "head-memory", "1Gi", "Amount of memory to use for the ray head. Default to 1Gi")
+	cmd.Flags().StringVar(&options.workerGrpName, "worker-grp-name", "default-group", "Name of the worker group for the Ray Cluster")
+	cmd.Flags().Int32Var(&options.workerReplicas, "worker-replicas", 1, "Number of the worker group replicas. Default of 1")
+	cmd.Flags().StringVar(&options.workerCPU, "worker-cpu", "500m", "Number of CPU for the ray worker. Default to 500m")
```

Comment:
Can we use a higher default, like 2 CPUs?

Reply:
Done!